### PR TITLE
1301 - OAuth2 flow with OpenID

### PR DESCRIPTION
**Pull request by orzechdev**
I want to merge this change because... it adds OAuth2 flow with OpenID support.

**PR intended to be tested with API branch:**  OAuth2_OpenID_flow
https://github.com/mirumee/saleor/pull/6182

### Screenshots

Zrzut ekranu 2021-01-26 o 23 11 01

### Pull Request Checklist

1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Translated strings are extracted.
1. [ ] Number of API calls is optimized.
1. [ ] The changes are tested.
1. [ ] Data-test are added for new elements.
1. [ ] Type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### Test environment config

API_URI=https://oauth2-openid-flow.api.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-1301 Implement sample OAuth flow for dashboard](https://app.clickup.com/t/2549495/SALEOR-1301)

File: src/auth/views/Login.tsx [hunk 1]
```diff
@@ -1,20 +1,89 @@
...
+    setIsExternalError(false);
+    setIsError(!result || errors?.length > 0);
+    return errors;
+  };
+
+  const handleRequestExternalAuthentication = (pluginId: string) =>
+    requestLoginByExternalPlugin(pluginId, {
+      redirectUri: urlJoin(
+        window.location.origin,
+        APP_MOUNT_URI === "/" ? "" : APP_MOUNT_URI,
```
Comment (mmarkusik) [inline src/auth/views/Login.tsx:51]: Can we put "/" to const like DEFAULT_URI ?

File: src/auth/views/Login.tsx [hunk 2]
```diff
@@ -1,20 +1,89 @@
...
+
+  const handleExternalAuthentication = async (code: string, state: string) => {
+    const result = await loginByExternalPlugin({ code, state });
+    const errors = result?.errors || [];
+
+    setIsError(false);
+    if (!result || errors?.length > 0) {
+      setIsExternalError(true);
+    } else {
+      navigate("/");
```
Comment (mmarkusik) [inline src/auth/views/Login.tsx:64]: Also use the above here

**Review by mmarkusik — APPROVED** [review]

File: src/auth/views/Login.tsx (hunk 1 above)
Comment (orzechdev) [reply src/auth/views/Login.tsx]: sure 👍

**Review by jwm0 — APPROVED** [review]

**Pull request merged**